pkg/view/share.go

```diff
@@ -0,0 +1,88 @@
+package view
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	"github.com/scratchdata/scratchdata/pkg/config"
+	"github.com/scratchdata/scratchdata/pkg/destinations"
+	"github.com/scratchdata/scratchdata/pkg/storage"
+)
+
+func RegisterShareView(
+	r *chi.Mux,
+	storageServices *storage.Services,
+	destinationManager *destinations.DestinationManager,
+	c config.DashboardConfig,
+) {
+	gv := newViewEngine(c.LiveReload)
+	r.Get("/share/{uuid}", func(w http.ResponseWriter, r *http.Request) {
+		queryUUID := chi.URLParam(r, "uuid")
+
+		id, err := uuid.Parse(queryUUID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		cachedQuery, found := storageServices.Database.GetShareQuery(r.Context(), id)
+		if !found {
+			http.Error(w, "Query not found", http.StatusNotFound)
+			return
+		}
+
+		year, month, day := cachedQuery.ExpiresAt.Date()
+
+		m := Model{
+			HideSidebar: true,
+			ShareQuery: ShareQuery{
+				Expires: fmt.Sprintf("%s %d, %d", month.String(), day, year),
+				ID:      id.String(),
+				Name:    cachedQuery.Name,
+			},
+		}
+		if err := gv.Render(w, http.StatusOK, "pages/share", m); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	})
+
+	r.Get("/share/{uuid}/download", func(w http.ResponseWriter, r *http.Request) {
+		format := strings.ToLower(r.URL.Query().Get("format"))
+
+		queryUUID := chi.URLParam(r, "uuid")
+
+		id, err := uuid.Parse(queryUUID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		cachedQuery, found := storageServices.Database.GetShareQuery(r.Context(), id)
```

Comment:
This code is copied from where it is used in API. As a part of the refactor, this functionality should be easier to reuse.